src/main/java/Parser.java

```diff
@@ -0,0 +1,121 @@
+import java.time.format.DateTimeParseException;
+import java.util.ArrayDeque;
+
+import javafx.scene.control.Label;
+
+/**
+ * Parser class for CS2103T iP. Handles the logic of the program.
+ */
+public class Parser {
+    private final TaskList list;
+    private final Ui ui;
+
+    /**
+     * Creates a Parser object.
+     * @param list TaskList object for data manipulation.
+     * @param ui Ui object for messages.
+     */
+    public Parser(TaskList list, Ui ui) {
+        this.list = list;
+        this.ui = ui;
+    }
+
+    /**
+     * Main logic of the program.
+     * @return Label with response from program.
```

Comment:
You might want to add a blank line in between the description and @return. Reference for code style here [https://se-education.org/guides/conventions/java/index.html#comments](https://se-education.org/guides/conventions/java/index.html#comments)

Similarly for all other javadocs